### PR TITLE
Implements wraapper for Application.OpenURL

### PR DESCRIPTION
**Pull request by Bod9001**
### Changelog:

CL: [New] Implements Secure wraapper for Application.OpenURL

### CODE REVIEW

**Comment by Bod9001** [comment]
if anyone can think of any more Checks to add that would be good

File: UnityProject/Assets/Mirror/Runtime/Transports/Ignorance/Editor/Toolbox.cs [hunk 1]
```diff
@@ -22,19 +23,19 @@ public static void RevealEnetLibraryName()
...
         {
-            UnityEngine.Application.OpenURL("https://github.com/SoftwareGuy/Ignorance/issues");
+              Debug.LogError("https://github.com/SoftwareGuy/Ignorance/issues");
         }
 
         [MenuItem("Ignorance/RTFM/ENet-C# Repo")]
         private static void LaunchENetCSharpForkRepo()
         {
-            UnityEngine.Application.OpenURL("https://github.com/SoftwareGuy/ENet-CSharp");
+            Debug.LogError("https://github.com/SoftwareGuy/ENet-CSharp");
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Mirror/Runtime/Transports/Ignorance/Editor/Toolbox.cs:38]: this doesn't sound like a good solution imo

File: UnityProject/Assets/Scripts/Core/SecureStuff/URLOpen/SafeURLOpen.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+using UnityEngine;
+
+public static class SafeURL
+{
+
+	public static void Open(string URL)
+	{
+		URL = URL.ToLower();
+
+		if (URL.StartsWith($"http://") || URL.StartsWith($"https://"))
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Core/SecureStuff/URLOpen/SafeURLOpen.cs]: should we prevent `http` requests? It's better if we force everything to use `https`

File: UnityProject/Assets/Mirror/Runtime/Transports/Ignorance/Editor/Toolbox.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Mirror/Runtime/Transports/Ignorance/Editor/Toolbox.cs:38]: idk, this just needed for the editor not sure if it's going to be in the actual ignorance assembly or in the main assets one,

File: UnityProject/Assets/Scripts/Core/SecureStuff/URLOpen/SafeURLOpen.cs (hunk 2 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/SecureStuff/URLOpen/SafeURLOpen.cs]: idk, anything that requires https will just redirect it

File: UnityProject/Assets/Scripts/Core/SecureStuff/URLOpen/SafeURLOpen.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+using UnityEngine;
+
+public static class SafeURL
+{
+
+	public static void Open(string URL)
+	{
+		URL = URL.ToLower();
+
+		if (URL.StartsWith($"https://"))
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/SecureStuff/URLOpen/SafeURLOpen.cs:15]: don't parse this yourself, use URIs instead. https://learn.microsoft.com/en-us/dotnet/api/system.uri?view=netframework-4.8.1

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/SecureStuff/URLOpen/SafeURLOpen.cs:15]: does later on in the function

Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/SecureStuff/URLOpen/SafeURLOpen.cs:15]: get and check if the protocol is valid using an URI, do not parse URLs into strings to do validation.

**Comment by Bod9001** [comment]
Continued in  #9999 due to git shenanigans

**Pull request merged**